Add one board Green Tea SPI Communication Test

mprse opened the pull request:
### Description

Provide one board SPI communication test which uses Green Tea Framework.

This PR can not be merged until PR which defines the new HAL API and adds at least one example implementation is merged. Probably then the test will have to be adapted to the final API version.

Detailed description can be found in README.md file in the test folder.

**List of PRs with related tests**

| PR | Description |
| --- | --- |
| RFC SPI: Reference Implementation. #8445 | New HAL API definition and example implementations for K66F anf NUCLEO_F429ZI |
| Add SPI test and test header file. #7976 | SPI Green Tea basic test |
| Add GreeTea SPI communication test. #8216 | Two boards Green Tea SPI communication test (closed) |
| Add Ice Tea SPI communication test. #8443 | Two boards  Ice Tea SPI communication test |
| Add one board Green Tea SPI Communication Test #8919  | One board Green Tea SPI communication test |

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [X] Test update
    [ ] Breaking change

mprse committed:
Add one board Green Tea SPI Communication Test

mprse commented:
Test results for the example implementation of SPI driver for NUCLEO_F429ZI:

![image](https://user-images.githubusercontent.com/30721012/49286563-d033d980-f49a-11e8-86ec-d171689dd3cb.png)

jamesbeyond commented:
Well done @mprse !

@screamerbg  or @MarceloSalazar need to have a look at this as well. please add them as reviewer.

screamerbg commented:
Thanks! I'm travelling today, but will try it tomorrow morning

jamesbeyond commented on TESTS/mbed_hal/spi_com/configs.h:
```diff
@@ -0,0 +1,27 @@
+#ifndef MBED_SPI_PINS_H
+#define MBED_SPI_PINS_H
+
+#if defined(TARGET_NUCLEO_F429ZI)
```
If every target has to add to this config file, and `pins.h`, is this file going to grow too big and a quite messy? Is there a better way to achieve the same function?

mprse replied:
We had a discussion with @ithinuel about this. I was thinking to use `app_config.json` config file. But as far as I remember @ithinuel pointed out that json is not used our green-tea tests and solution with header files is already used in QSPI tests. This is why we decided to use this solution.

cmonr replied:
Huh. @ARMmbed/mbed-os-core Any thoughts on this?

mprse replied:
I discussed this with @maciejbocianski and QSPI test doesn't use header files for pins definition, but only for flash config. In QSPI pins have been unified across all targets with QSPI feature enabled. So in this case we could also define SPI pins in target's `PinNames.h` file. To run one board test we require two SPI interfaces, so I suggest to add the following definitions:
`SPI0_MOSI`, `SPI0_MISO`, `SPI0_SCK`, `SPI0_CS`
`SPI1_MOSI`, `SPI1_MISO`, `SPI1_SCK`, `SPI1_CS`

cc @c1728p9 @0xc0170 @donatieng @jamesbeyond

0xc0170 replied:
Looks better than these pins in the header file, to define SPIx pins

jamesbeyond replied:
This sounds a better idea to me!

mprse replied:
Test has been updated:
- removed header files with pins definition
- provided pins definition in target's `PinNames.h` file
- adapted pin names in the test
- adapted documentation.

cmonr commented:
@0xc0170 @jamesbeyond @MarceloSalazar @screamerbg When y'all get a chance.

cmonr commented:
Depends on _at least_ https://github.com/ARMmbed/mbed-os/pull/8445

cmonr commented:
@mprse Does this depend on anything else aside for #8445?

mprse commented:
I need to clean up spi pins definiotion and probably I will remove the definitions from this PR and add them to PR https://github.com/ARMmbed/mbed-os/pull/7976. So they are consistent. Then #7976 will have to go first.

mprse commented:
This test is ready for review, but requires SPI pins definition from #7976.

mprse commented:
Rebased after spi feature branch has been rebased on master to solve Travis issue.

mprse commented:
Rebased again after spi feature branch has been rebased on master to solve second Travis issue.
This PR require PR https://github.com/ARMmbed/mbed-os/pull/7976 to be merged first.

mprse commented:
PR #7976 has been merged. I have rebased this one it is ready for review and CI.

0xc0170 commented:
Reviewes are opened 🎉

mprse committed:
ARM_CM3DS_MPS2: Disable ANALOGIN support

ANALOGIN is related to SPI. If SPI is not supported, then ANALOGIN also.

This change should fix:
ARM_CM3DS_MPS2::ARMC6::TESTS-MBED_HAL-PINMAP
[Error] @0,0: L6218E: Undefined symbol analogin_pinmap (referred from BUILD/tests/ARM_CM3DS_MPS2/ARMC6/TESTS/mbed_hal/pinmap/TESTS/mbed_hal/pinmap/main.o).

mprse committed:
FVP_MPS2 family: Disable SERIAL_FC

Looks like `FVP_MPS2 family` has SERIAL_FC feature enabled, but does not provide pins definition for `PinMap_UART_CTS`, `PinMap_UART_RTS`.

This should fix the following failures:
ARMC6::FVP_MPS2_M0
ARMC6::FVP_MPS2_M0P
ARMC6::FVP_MPS2_M3
ARMC6::FVP_MPS2_M4
ARMC6::FVP_MPS2_M7
[Error] serial_api.c@403,12: use of undeclared identifier 'PinMap_UART_CTS'; did you mean 'PinMap_UART_TX'?

mprse committed:
Disable mbed-os-example-attestation, mbed-os-example-blockdevice examples.

Not supported on the spi feature branch.

mprse committed:
K64F: Adapt spi driver to the new HAL API, enable SPI
